### PR TITLE
Native queue length report with metadata report (version 2.0)

### PR DESCRIPTION
**Pull request by tmasternak**
This is core v6 version of #34

### CODE REVIEW

File: src/NServiceBus.Metrics.ServiceControl.AcceptanceTests/Tests/QueueLength/When_native_queue_length_is_reported.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+        public class RawMessageBehavior : Behavior<IIncomingPhysicalMessageContext>
+        {
+            public Context TestContext { get; set; }
+
+            public override Task Invoke(IIncomingPhysicalMessageContext context, Func<Task> next)
+            {
+                if (context.MessageHeaders.TryGetValue("NServiceBus.Metric.Type", out var metricType) && metricType == "QueueLength")
+                {
+                    TestContext.QueueLengthReportReceived = true;
```
Comment (Scooletz) [inline src/NServiceBus.Metrics.ServiceControl.AcceptanceTests/Tests/QueueLength/When_native_queue_length_is_reported.cs]: Why don't we assert for the values being sent in here? How are we sure that "queue" and 10 are being sent? Are we do it on the basis of the buffer underneath?

File: src/NServiceBus.Metrics.ServiceControl/IReportNativeQueueLength.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        /// <summary>
+        /// Reports the length of a queue to be sent to ServiceControl Monitoring
+        /// </summary>
+        void ReportQueueLength(string physicalQueueName, long queueLength);
+
+        /// <summary>
+        /// A list of queues that are monitored
+        /// </summary>
+        IEnumerable<string> MonitoredQueues { get; }
```
Comment (Scooletz) [inline src/NServiceBus.Metrics.ServiceControl/IReportNativeQueueLength.cs]: What is this collection for? It's exposed and never used. It looks like that was attached to the reporter interface.

File: src/NServiceBus.Metrics.ServiceControl/ServiceControlReporting/EndpointMetadata.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        public EndpointMetadata(string localAddress)
+        {
+            this.localAddress = localAddress;
+        }
+
+        public string ToJson()
+        {
+            return SimpleJson.SerializeObject(new
+            {
+                Version = 2,
```
Comment (Scooletz) [inline src/NServiceBus.Metrics.ServiceControl/ServiceControlReporting/EndpointMetadata.cs]: Why is this `2`? No comment about the version value.

File: src/NServiceBus.Metrics.ServiceControl.AcceptanceTests/Tests/QueueLength/When_native_queue_length_is_reported.cs (hunk 1 above)
Comment (tmasternak) [reply src/NServiceBus.Metrics.ServiceControl.AcceptanceTests/Tests/QueueLength/When_native_queue_length_is_reported.cs]: Assetions added.

File: src/NServiceBus.Metrics.ServiceControl/IReportNativeQueueLength.cs (hunk 2 above)
Comment (tmasternak) [reply src/NServiceBus.Metrics.ServiceControl/IReportNativeQueueLength.cs]: This is a collection of queues that endpoint level monitor is supposed to track. For now it holds only a single value equal to LocalAddress

File: src/NServiceBus.Metrics.ServiceControl/ServiceControlReporting/EndpointMetadata.cs (hunk 3 above)
Comment (tmasternak) [reply src/NServiceBus.Metrics.ServiceControl/ServiceControlReporting/EndpointMetadata.cs]: Name changed to `PluginVersion`

**Comment by tmasternak** [comment]
@Scooletz review changes applied. I'm not including the `Breaking changes` section as it's in the connected PR for version 1.

File: src/NServiceBus.Metrics.ServiceControl/IReportNativeQueueLength.cs (hunk 2 above)
Comment (Scooletz) [reply src/NServiceBus.Metrics.ServiceControl/IReportNativeQueueLength.cs]: I had a conversation with @tmasternak and we agreed that the current interface is just fine. Let's move on with it.

**Pull request closed without merging**